core/src/miner/mem_pool.rs

```diff
@@ -934,7 +934,105 @@ impl MemPool {
+                .current
+                .queue
+                .iter()
+                .map(|t| {
+                    self.by_hash
+                        .get(&t.hash)
+                        .expect("All transactions in `current` and `future` are always included in `by_hash`")
+                })
+                .filter(|t| range.contains(&t.inserted_timestamp))
+                .count()
+    }
+
+    /// Return all future transactions along with current transactions.
+    pub fn get_future_pending_transactions(
+        &self,
+        size_limit: usize,
+        current_timestamp: Option<u64>,
+        range: Range<u64>,
+    ) -> PendingSignedTransactions {
+        let mut current_size: usize = 0;
+        let future_pending_items: Vec<_> = self
+            .future
+            .queue
+            .iter()
+            .map(|t| {
+                self.by_hash
+                    .get(&t.hash)
+                    .expect("All transactions in `current` and `future` are always included in `by_hash`")
+            })
+            .filter(|t| {
+                if let Some(expiration) = t.expiration() {
+                    if let Some(timestamp) = current_timestamp {
+                        return expiration >= timestamp
+                    }
+                }
+                true
+            })
+            .filter(|t| range.contains(&t.inserted_timestamp))
+            .take_while(|t| {
+                let encoded_byte_array = rlp::encode(&t.tx);
+                let size_in_byte = encoded_byte_array.len();
+                current_size += size_in_byte;
+                current_size < size_limit
+            })
+            .collect();
+
+
+        let pending_items: Vec<_> = self
+            .current
+            .queue
+            .iter()
+            .map(|t| {
+                self.by_hash
+                    .get(&t.hash)
+                    .expect("All transactions in `current` and `future` are always included in `by_hash`")
+            })
+            .filter(|t| {
+                if let Some(expiration) = t.expiration() {
+                    if let Some(timestamp) = current_timestamp {
+                        return expiration >= timestamp
+                    }
+                }
+                true
+            })
+            .filter(|t| range.contains(&t.inserted_timestamp))
+            .take_while(|t| {
+                let encoded_byte_array = rlp::encode(&t.tx);
+                let size_in_byte = encoded_byte_array.len();
+                current_size += size_in_byte;
+                current_size < size_limit
+            })
+            .collect();
+
+
+        let mut current_signed_tx: Vec<SignedTransaction> = pending_items.iter().map(|t| t.tx.clone()).collect();
+        let mut future_signed_tx: Vec<SignedTransaction> = future_pending_items.iter().map(|t| t.tx.clone()).collect();
+        for tx in &mut future_signed_tx {
+            current_signed_tx.push(tx.clone());
+        }
```

Comment:
I think there is a better way to concatenating 2 vectors. but, I'm not sure at this moment.

You can look at https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.chain Or ask anybody else.
These days I don't write Rust code much. 😓